src/engine.rs

```diff
@@ -136,8 +136,13 @@ where
     /// bytes. If `sync` is true, the write will be followed by a call to
     /// `fdatasync` on the log file.
     pub fn write(&self, log_batch: &mut LogBatch, mut sync: bool) -> Result<usize> {
+        if log_batch.is_empty() {
+            return Ok(0);
+        }
         let start = Instant::now();
         let len = log_batch.finish_populate(self.cfg.batch_compression_threshold.0 as usize)?;
+        debug_assert!(!log_batch.is_empty());
```

Comment:
Do it before `finish_populate` or remove it. The first validation `log_batch.is_empty()` has guaranteed that `log_batch.is_empty() == false`.